### PR TITLE
feat: Configurable API server timeout

### PR DESCRIPTION
**Pull request by imranismail**
## Proposed changes

Allows API server timeout to be configured

## Checklist

- [X] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md).
- [X] I have read the [security policy](../security/policy).
- [X] I confirm that this pull request does not address a security
      vulnerability. If this pull request addresses a security. vulnerability, I
      confirm that I got green light (please contact
      [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push
      the changes.
- [ ] I have added tests that prove my fix is effective or that my feature
      works.
- [ ] I have added or changed [the documentation](docs/docs).

### CODE REVIEW

**Comment by aeneasr** [comment]
This looks good to me!

File: driver/configuration/provider_viper.go [hunk 1]
```diff
@@ -173,6 +176,18 @@ func (v *ViperProvider) ProxyServeAddress() string {
 	)
 }
 
+func (v *ViperProvider) APIReadTimeout() time.Duration {
+	return viperx.GetDuration(v.l, ViperKeyAPIReadTimeout, time.Second*5, "API_SERVER_READ_TIMEOUT")
```
Comment (aeneasr) [inline driver/configuration/provider_viper.go:180]:
```suggestion
	return viperx.GetDuration(v.l, ViperKeyAPIReadTimeout, time.Second*5)
```

File: driver/configuration/provider_viper.go [hunk 2]
```diff
@@ -173,6 +176,18 @@ func (v *ViperProvider) ProxyServeAddress() string {
...
+}
+
+func (v *ViperProvider) APIWriteTimeout() time.Duration {
+	return viperx.GetDuration(v.l, ViperKeyAPIWriteTimeout, time.Second*10, "API_SERVER_WRITE_TIMEOUT")
```
Comment (aeneasr) [inline driver/configuration/provider_viper.go:184]:
```suggestion
	return viperx.GetDuration(v.l, ViperKeyAPIWriteTimeout, time.Second*10)
``

File: driver/configuration/provider_viper.go [hunk 3]
```diff
@@ -173,6 +176,18 @@ func (v *ViperProvider) ProxyServeAddress() string {
...
+}
+
+func (v *ViperProvider) APIIdleTimeout() time.Duration {
+	return viperx.GetDuration(v.l, ViperKeyAPIIdleTimeout, time.Second*120, "API_SERVER_IDLE_TIMEOUT")
```
Comment (aeneasr) [inline driver/configuration/provider_viper.go:188]:
```suggestion
	return viperx.GetDuration(v.l, ViperKeyAPIIdleTimeout, time.Second*120)
```

File: .schema/config.schema.json [hunk 4]
```diff
@@ -982,6 +982,50 @@
               "title": "Host",
               "description": "The network interface to listen on."
             },
+            "timeout": {
```
Comment (aeneasr) [inline .schema/config.schema.json:985]: If you take [this part](https://github.com/ory/oathkeeper/blob/e25501c1c39ec2f6fa93f72cd59ec571fd105f86/.schema/config.schema.json#L985) and make it a definition [(example)](https://github.com/ory/oathkeeper/blob/e25501c1c39ec2f6fa93f72cd59ec571fd105f86/.schema/config.schema.json#L7-L22) called `serverTimeout`, you can reference the definition both here and [there](https://github.com/ory/oathkeeper/blob/e25501c1c39ec2f6fa93f72cd59ec571fd105f86/.schema/config.schema.json#L985) using `#/definitions/serverTimeout`

File: .schemas/config.schema.json [hunk 5]
```diff
@@ -922,6 +922,50 @@
               "title": "Host",
               "description": "The network interface to listen on."
             },
+            "timeout": {
```
Comment (aeneasr) [inline .schemas/config.schema.json:925]: This directory is no longer used - see the README :)

Comment (imranismail) [reply .schemas/config.schema.json:925]: Where in the readme is this mentioned?

Comment (aeneasr) [reply .schemas/config.schema.json:925]: I should have been more specific, this one: https://github.com/ory/oathkeeper/blob/master/.schemas/README.md

File: driver/configuration/provider_viper.go [hunk 6]
```diff
@@ -173,6 +176,19 @@ func (v *ViperProvider) ProxyServeAddress() string {
 	)
 }
 
+func (v *ViperProvider) APIReadTimeout() time.Duration {
+	return viperx.GetDuration(v.l, ViperKeyAPIReadTimeout, time.Second*5)
+}
+
+func (v *ViperProvider) APIWriteTimeout() time.Duration {
+	return viperx.GetDuration(v.l, ViperKeyAPIWriteTimeout, time.Second*10)
+``
```
Comment (aeneasr) [inline driver/configuration/provider_viper.go:185]: Looks like this somehow made it in here :D

**Comment by aeneasr** [comment]
Thank you!

**Pull request merged**